packages/react-instantsearch-hooks-dom/src/ui/Breadcrumb.tsx

```diff
@@ -47,12 +47,17 @@ type UseBreadcrumbRenderState = ReturnType<typeof useBreadcrumb>;
 
 type BreadcrumbItem = UseBreadcrumbRenderState['items'][number];
 
-export type BreadcrumbProps = React.HTMLAttributes<HTMLDivElement> &
+type WrapperProps = React.ComponentProps<'div'>;
+
+export type BreadcrumbWidgetProps = WrapperProps & {
+  classNames?: Partial<BreadcrumbClassNames>;
+};
+
+export type BreadcrumbProps = BreadcrumbWidgetProps &
```

Comment:
Why do we have all this indirection types and not this?

```tsx
export type BreadcrumbProps = React.ComponentProps<'div'> &
  Pick<UseBreadcrumbRenderState, 'items' | 'createURL'> & {
    hasItems: boolean;
    onNavigate: UseBreadcrumbRenderState['refine'];
    separator: string | undefined;
    translations: BreadcrumbTranslations;
    classNames?: Partial<BreadcrumbClassNames>;
  };
```

Reply:
in the widget, how would you then decide which props to expose from the users' POV?